ui/imports/shared/views/chat/MessageView.qml

```diff
@@ -1061,25 +1066,22 @@ Loader {
                 }
 
                 quickActions: [
-                    Loader {
-                        active: d.addReactionAllowed && delegate.hovered && !root.isViewMemberMessagesePopup
-
-                        visible: active
-                        sourceComponent: StatusFlatRoundButton {
-                            width: d.chatButtonSize
-                            height: d.chatButtonSize
-                            icon.name: "reaction-b"
-                            type: StatusFlatRoundButton.Type.Tertiary
-                            enabled: !root.emojiReactionLimitReached
-                            tooltip.text: {
-                                if (root.emojiReactionLimitReached) {
-                                    return qsTr("Maximum number of different reactions reached")
-                                }
-                                return qsTr("Add reaction")
-                            }
-                            onClicked: (mouse) => {
-                                d.addReactionClicked(this, mouse)
-                            }
+                    MessageReactionsRow {
+                        visible: {
+                            root.emojiReactionLimitReached
+                            return !root.emojiReactionLimitReached && !root.isViewMemberMessagesePopup
+                        }
+                        buttonSize: d.chatButtonSize
+                        leftPadding: 0
+                        rightPadding: 0
+                        emojiModel: emojiPopup.emojiModel
+                        recentEmojis: emojiPopup.recentEmojis
+                        skinColor: emojiPopup.skinColor
+                        onToggleReaction: emoji => {
+                            root.messageStore.toggleReaction(root.messageId, emoji)
```

Comment:
Probably this view has already some direct calls to the store but if we can avoid new ones, should be nice! I'd propose to create a `toggleReaction` signal and move the store call up to the layout or a more high level component.